### PR TITLE
Add instructions for dealing with systemd PrivateNetwork

### PR DESCRIPTION
**Pull request by DrDaveD**

### CODE REVIEW

File: docs/worker-node/install-singularity.md [hunk 1]
```diff
@@ -140,6 +140,24 @@ unprivileged.
     such as Docker.  Disabling network namespaces may break other
     software, or limit its capabilities (such as requiring the
     `--net=host` option in Docker).
+    
+    Disabling network namespaces blocks the systemd PrivateNetwork
+    feature, which no default enabled EL 7 services depend on but some
+    EL 8 services do.  To check them all, look for PrivateNetwork in
```
Comment (brianhlin) [inline docs/worker-node/install-singularity.md:146]: It took me a couple of tries to parse the second clause of this sentence.  Maybe just simplify it by saying that it's a feature more commonly seen across EL 8 services?

File: docs/worker-node/install-singularity.md [hunk 2]
```diff
@@ -140,6 +140,24 @@ unprivileged.
...
+    `/lib/systemd/system/*.service` and see which of those services are
+    enabled but failed to start.  The only default such service on EL 8
+    is systemd-hostnamed, and a popular non-default such service is
+    mlocate-updatedb.  The PrivateNetwork feature can be turned off for
+    a service without modifying an rpm-installed file through a
```
Comment (brianhlin) [inline docs/worker-node/install-singularity.md:151]:
```suggestion
    a service without modifying an RPM-installed file through a
```

File: docs/worker-node/install-singularity.md [hunk 3]
```diff
@@ -140,6 +140,24 @@ unprivileged.
...
+    `<service>.d/*.conf` file, for example for systemd-hostnamed:
+
+        :::console
+        root@host # cd /lib/systemd/system
```
Comment (brianhlin) [inline docs/worker-node/install-singularity.md:155]:
```suggestion
        root@host # cd /etc/systemd/system
```

I would think that admin changes would go into `/etc`

**Review by brianhlin — APPROVED** [review]
Some minor comments/suggestions. Otherwise LGTM

File: docs/worker-node/install-singularity.md (hunk 1 above)
Comment (DrDaveD) [reply docs/worker-node/install-singularity.md:146]: I actually see it in several systemd files on EL 7 but they're all disabled services.  I'll spell that out in two sentences.

File: docs/worker-node/install-singularity.md [hunk 4]
```diff
@@ -140,6 +140,25 @@ unprivileged.
...
+    mlocate-updatedb.  The PrivateNetwork feature can be turned off for
+    a service without modifying an RPM-installed file through a
+    `<service>.d/*.conf` file, for example for systemd-hostnamed:
+
+        :::console
+        root@host # cd /etc/systemd/system
+        root@host # mkdir -p systemd-hostnamed.service.d
+        root@host # (echo "[Service]"; echo "PrivateNetwork=no") \
+                >systemd-hostnamed.service.d/no-private-network.conf
+        root@host # systemctl start systemd-hostnamed
```
Comment (brianhlin) [inline docs/worker-node/install-singularity.md:160]:
```suggestion
        root@host # systemctl daemon-reload
        root@host # systemctl start systemd-hostnamed
```

Personally I always do this after making changes but I'm not sure if it's necessary

**Review by brianhlin — APPROVED** [review]
One last suggestion but otherwise LGTM!

File: docs/worker-node/install-singularity.md (hunk 4 above)
Comment (DrDaveD) [reply docs/worker-node/install-singularity.md:160]: Yes it is necessary, thanks for catching it.

**Pull request merged**